lib/csu/common-cheri/crt_init_globals.c

```diff
@@ -175,7 +175,8 @@ do_crt_init_globals(const Elf_Phdr *phdr, long phnum)
 			/* TODO: should we allow a single RWX segment? */
 			__builtin_trap();
 		}
-		data_cap = cheri_setaddress(phdr, writable_start);
+		data_cap = cheri_setaddress(
+		    __DECONST(Elf_Phdr *, phdr), writable_start);
```

Comment:
In this case we don't _need_ the type and could just use `(void *)`, since that's what `data_cap`'s type is.